### PR TITLE
More gen 10: Gen favorites

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Gen favorites

### CODE REVIEW

File: shared/actions/gregor.js [hunk 1]
```diff
@@ -170,17 +169,18 @@ function* handleKbfsFavoritesOOBM(kbfsFavoriteMessages: Array<OutOfBandMessage>)
   const msgsWithParsedBodies = kbfsFavoriteMessages.map(m => ({...m, body: JSON.parse(m.body.toString())}))
   const createdTLFs = msgsWithParsedBodies.filter(m => m.body.action === 'create')
 
-  const username: string = (yield select(usernameSelector): any)
+  const username: string = (yield Saga.select(usernameSelector): any)
   const folderActions = createdTLFs.reduce((arr, m) => {
-    const folder = m.body.tlf ? markTLFCreated(folderFromPath(username, m.body.tlf)) : null
```
Comment (chrisnojima) [inline shared/actions/gregor.js]: I changed this slightly so we make the action after we evaluate that its good

File: shared/reducers/favorite.js [hunk 2]
```diff
@@ -1,52 +1,20 @@
 // @flow
 import * as Constants from '../constants/favorite'
+import * as FavoriteGen from '../actions/favorite-gen'
 import * as KBFSGen from '../actions/kbfs-gen'
 
-const initialState: Constants.State = {
```
Comment (chrisnojima) [inline shared/reducers/favorite.js]: move this into constants

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
Thanks, i (annoyingly) fixed this in the most recent pr: https://github.com/keybase/client/pull/9549/commits/47fd0e6eb72f66cc4b9b6af102c4e3a64d3ebcdd

File: shared/actions/favorite.js [hunk 3]
```diff
@@ -187,42 +163,30 @@ function _getFavoritesRPCToFolders(
...
+      yield put(FavoriteGen.createFavoriteAdded())
+      yield put(FavoriteGen.createFavoriteList())
     } catch (error) {
       console.warn('Err in favorite.favoriteAdd', error)
     }
   }
 }
 
-function* _ignoreSaga(action: Constants.FavoriteAdd): Saga.SagaGenerator<any, any> {
+function* _ignoreSaga(action: FavoriteGen.FavoriteAddPayload): Saga.SagaGenerator<any, any> {
```
Comment (buoyad) [inline shared/actions/favorite.js]: Should this be `FavoriteIgnorePayload`?

Comment (chrisnojima) [reply shared/actions/favorite.js]: yup fixed tnx!

**Pull request merged**